### PR TITLE
IGNITE-18254: Sql. Extend SQL grammar with ALTER ZONE statement

### PR DESCRIPTION
**Pull request by lowka**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlAlterZoneRenameTo.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    /**
+     * The new name for a distribution zone.
+     */
+    public SqlIdentifier newName() {
+        return newName;
+    }
+
+
```
Comment (zstan) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlAlterZoneRenameTo.java:61]: extra line

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteAbstractSqlAlterZone.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    /** Alter operator. */
+    private static final SqlOperator OPERATOR =
+            new SqlSpecialOperator("ALTER ZONE", SqlKind.OTHER_DDL);
+
+    /** Scope. **/
+    private static final String SCOPE = "ZONE";
+
+    /** Constructor. */
+    public IgniteAbstractSqlAlterZone(SqlParserPos pos) {
```
Comment (zstan) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteAbstractSqlAlterZone.java:39]: seems **public** access is redundant here

**Review by zstan — APPROVED** [review]
looks good, plz fix minor issues and lets move forward.

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 3]
```diff
@@ -532,3 +532,96 @@ SqlDrop SqlDropZone(Span s, boolean replace) :
...
+SqlNodeList AlterZoneOptions() :
+{
+  List<SqlNode> list = new ArrayList<SqlNode>();
+  final Span s = Span.of();
+}
+{
+  AlterZoneOption(list)
+  (
+      <COMMA> { s.add(this); } AlterZoneOption(list)
+  )+
```
Comment (korlov42) [inline modules/sql-engine/src/main/codegen/includes/parserImpls.ftl:566]: did you mean `*`? With current rule it's impossible to change exactly one param. Also, we need to cover this case in tests too

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 4]
```diff
@@ -532,3 +532,96 @@ SqlDrop SqlDropZone(Span s, boolean replace) :
...
+}
+{
+    <ALTER> { s = span(); }
+    <ZONE> zoneId = CompoundIdentifier()
+    (
+      <RENAME> <TO> newZoneId = CompoundIdentifier() {
+        return new IgniteSqlAlterZoneRenameTo(s.end(this), zoneId, newZoneId);
+      }
+      |
+      <WITH> { s.add(this); } optionList = AlterZoneOptions() {
```
Comment (korlov42) [inline modules/sql-engine/src/main/codegen/includes/parserImpls.ftl:551]: IMO, we should use `WITH` keyword to provide context of the operation. To alter a certain property `SET` seems more natural to me. WDYT?

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteAbstractSqlAlterZone.java (hunk 2 above)
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteAbstractSqlAlterZone.java:39]: Fixed

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlAlterZoneRenameTo.java (hunk 1 above)
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlAlterZoneRenameTo.java]: fixed

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl (hunk 4 above)
Comment (lowka) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl:551]: Sure.

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 5]
```diff
@@ -532,3 +532,96 @@ SqlDrop SqlDropZone(Span s, boolean replace) :
...
+SqlNode SqlAlterZone() :
+{
+    final Span s;
+    final SqlIdentifier zoneId;
+    final SqlIdentifier newZoneId;
+    SqlNodeList optionList = null;
+}
+{
+    <ALTER> { s = span(); }
+    <ZONE> zoneId = CompoundIdentifier()
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl:545]: The ticket does mention such syntax but sure, no problem.

File: modules/sql-engine/src/main/codegen/config.fmpp [hunk 6]
```diff
@@ -29,6 +29,7 @@ data: {
     imports: [
       "org.apache.calcite.sql.SqlCreate",
       "org.apache.calcite.sql.SqlDrop",
+      "org.apache.calcite.sql.SqlDdl",
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/codegen/config.fmpp]: fixed

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 7]
```diff
@@ -532,3 +532,96 @@ SqlDrop SqlDropZone(Span s, boolean replace) :
...
+    (
+      <RENAME> <TO> newZoneId = CompoundIdentifier() {
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl]: Fixed

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl (hunk 3 above)
Comment (lowka) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl]: fixed

File: modules/sql-engine/src/main/codegen/config.fmpp [hunk 8]
```diff
@@ -594,7 +598,8 @@ data: {
     # Return type of method implementation should be 'SqlNode'.
     # Example: "SqlShowDatabases()", "SqlShowTables()".
     statementParserMethods: [
-      "SqlAlterTable()"
+      "SqlAlterTable()",
+      "SqlAlterZone()"
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/codegen/config.fmpp:601]: AlterZone* now extends SqlDdl as AlterTable*.

**Review by korlov42 — APPROVED** [review]

**Pull request merged**